### PR TITLE
Refactor `purge_nonempty_nulls`

### PR DESCRIPTION
**Pull request by ttnghia**
This refactor combines the discrete interfaces of `purge_nonempty_nulls` that require `structs/strings/lists_column_view` input into just one interface accepting just `column_view`. This facilitates easier usage of this function. It is also a necessary step for subsequent work in fixing `structs::superimpose_parent_nulls`.

`cudf::detail` interface for this new API is also added.

### CODE REVIEW

File: cpp/src/copying/purge_nonempty_nulls.cu [hunk 1]
```diff
@@ -80,6 +80,24 @@ bool has_nonempty_nulls(cudf::column_view const& input, rmm::cuda_stream_view st
 
   return false;
 }
+
+std::unique_ptr<column> purge_nonempty_nulls(column_view const& input,
```
Comment (ttnghia) [inline cpp/src/copying/purge_nonempty_nulls.cu:84]: This is moved from the deleted file `copy.cuh`.

**Review by davidwendt — APPROVED** [review]
LGTM

**Comment by ttnghia** [comment]
Rerun tests.

**Review by bdice — APPROVED** [review]
One minor comment on docstrings, otherwise LGTM.

Someday I'd like to see the "pass-through" version of this which accepts a `column &&` (takes ownership) and returns the existing data when possible instead of copying.

File: cpp/include/cudf/copying.hpp [hunk 2]
```diff
@@ -1012,12 +1012,19 @@ bool has_nonempty_nulls(column_view const& input);
...
+ * descendants.
  *
- * LIST columns may have non-empty null rows.
- * For example:
- * @code{.pseudo}
+ * If the input column is not of compound type (LIST/STRING/STRUCT/DICTIONARY), the output will be
+ * the same as input.
+ *
+ * The purge operation only applies directly to LIST and STRING columns, but it applies indirectly
+ * to STRUCT/DICTIONARY columns as well, since these columns may have child/descendant columns that
```
[the remark this answers is not in the document]
Comment (bdice) [reply cpp/include/cudf/copying.hpp]: I think it’s implied that the relationship is recursive. But this isn’t a major issue either way.

**Comment by ttnghia** [comment]
Rerun tests.

**Comment by ttnghia** [comment]
Rerun tests.

**Comment by ttnghia** [comment]
@gpucibot merge

**Review by ajschmidt8 — APPROVED** [review]
Approving `ops-codeowner` file changes

**Review by raydouglass — APPROVED** [review]

**Pull request merged**